src/server/routes.js

```diff
@@ -13,21 +16,80 @@ app.use(function (req, res, next) {
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 app.use(express.static(__dirname + "/../client/dist"));
+app.use(
+  cors({
+    origin: "*",
+  })
+);
+
 app.use(
   "/shop/:productId/:styleId",
   express.static(__dirname + "./../client/dist")
 );
 
-app.get("/inventory/:productid/:styleid", (req, res) => {
-  inventoryDB
-    .findStyle(req.params.productid, req.params.styleid)
-    .then((result) => {
+app.get("/inventory/:productID/:styleID", function (req, res) {
+  const pID = req.params.productID;
+  const sID = req.params.styleID;
+
+  controller
+    .read(pID, sID)
+    .then(function (result) {
+      res.send(result).status(200);
+    })
+    .catch(function (err) {
+      console.error(`Error: ${err}`);
+      res.sendStatus(500);
+    });
+});
+
+app.post("/inventory/product/add", function (req, res) {
+  const document = req.body;
+
+  controller
+    .write(document)
+    .then(function (result) {
+      res.send(result).status(200);
+    })
+    .catch(function (err) {
+      console.error(`Error: ${err}`);
+      res.sendStatus(500);
+    });
+});
+
+app.patch("/inventory/:productID/:styleID/update", function (req, res) {
+  const size = req.body.size;
+  const newQuantity = req.body.newQuantity;
+  const pID = req.params.productID;
+  const sID = req.params.styleID;
+
+  // Development use only console.log.
+  console.log(req.body);
+
+  controller
+    .updateQuantity(pID, sID, size, newQuantity)
+    .then(function (result) {
+      res.send(result).status(200);
+    })
+    .catch(function (err) {
+      console.error(`Error: ${err}`);
+      res.sendStatus(500);
+    });
+});
+
+app.delete("/inventory/:productID/:styleID/:size/delete", function (req, res) {
+  const pID = req.params.productID;
+  const sID = req.params.styleID;
+  const size = req.params.size;
```

Comment:
Do these need to be integers? This applies to the rest of the file too.

Reply:
Rationale: This is a specific design choice. Utilizing integers as product IDs and style IDs is unwise, as there is a limit the size of an integer storable within any selected database. 

As stated in this [documentation](https://www.postgresql.org/docs/9.1/datatype-numeric.html), the largest possible storage for integers is big serial. Big serial allows for 8 bytes of storage, which translates to 1 to 9223372036854775807 in terms of allowed integers. 

If you have any more unique identifies than that, you will no longer be able to store any new identifies. 

Additionally, this design is flexible enough to support IDs with characters in it, such as "AA123" or "123E4L5FFP". Real world application includes: Hedgefunds, which are required to store data that may exceed the quintillion count, as they must have access to old transaction histories as viable data for their models. 

Mixing strings and numbers as a product ID also provide a larger range than simply recording IDs from 1 to 9223372036854775807.

Reply:
Realistically speaking however, I don't think that we would have more than 10 billion items in an inventory. Also, correct me if I'm wrong, I'm pretty sure a string takes up more bytes the moment it gets past 4 characters (which would be 4 bytes for the binary, then 1 for each character) so the moment you have more than 9999 items, you save 1 byte for every integer ID instead if a bigint is used in this example. You then also get the capability of auto-incrementing stead of having to rely on logic on the code that I believe will more likely break (assuming the IDs are indeed only going to be numbers).

Maybe if we weren't talking about inventory count and more about authentication, I would see the use case for having a quintillion amount of different variations.

Reply:
Aside from that, I don't see an issue with doing IDs this way. I just wouldn't go as far as saying that using integers is unwise.

Reply:
My apologies, I should've worded it better. But I definitely see your point in why it won't be necessary for the scope of this sprint. It's also amazing you took note of this as this is a more subtle detail that can be overlooked. Your honest in-depth review and analysis is great for novice engineers like me; thanks!

Reply:
I would even go as far as saying even outside of this sprint. 9223372036854775807 is a really really large number to even hit. By the time Amazon takes to even get to that amount of transactions, I would say technology would already be by far much more advanced.

And we are both novice.